summoner-cli/examples/cabal-full/.github/workflows/ci.yml

```diff
@@ -7,39 +7,59 @@ on:
     branches: [master]
 
 jobs:
-  build:
-    name: ghc ${{ matrix.ghc }}
-    runs-on: ubuntu-16.04
+  cabal:
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
         cabal: ["2.4"]
         ghc:
           - "8.4.4"
           - "8.6.5"
           - "8.8.4"
           - "8.10.1"
+        exclude:
+          - os: macOS-latest
+            ghc: 8.4.4
+          - os: macOS-latest
+            ghc: 8.6.5
+          - os: macOS-latest
+            ghc: 8.8.4
+          - os: windows-latest
+            ghc: 8.4.4
+          - os: windows-latest
+            ghc: 8.6.5
+          - os: windows-latest
+            ghc: 8.8.4
 
     steps:
     - uses: actions/checkout@v2
       if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
 
-    - uses: actions/setup-haskell@v1
+    - uses: actions/setup-haskell@v1.1.1
+      id: setup-haskell-cabal
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
+    - name: Freeze
+      run: |
+        cabal v2-freeze
+
     - uses: actions/cache@v1
       name: Cache ~/.cabal/store
       with:
-        path: ~/.cabal/store
-        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal
+        path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+        key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
 
     - name: Build
       run: |
-        cabal v2-update
-        cabal v2-build --enable-tests --enable-benchmarks
+        cabal v2-configure --enable-tests --enable-benchmarks
+        cabal v2-build all
 
     - name: Test
       run: |
-        cabal v2-test --enable-tests --test-show-details=direct
```

Comment:
`--test-show-details=direct` didn't work for me with Cabal 2.4 (see https://github.com/gdziadkiewicz/ubiquitous-octo-umbrella/runs/1237695655?check_suite_focus=true#step:6:4 ).

Reply:
Yes, this is a known problem. This option is added only in `master`, not in the previous release. We were planning to move to Cabal-3.0 in the next release, so this will work eventually: https://github.com/kowainik/summoner/issues/340

Reply:
For now, it's okay to keep `--test-show-details` as it is :slightly_smiling_face:

Reply:
`--test-show-details=direct` has been restored in `cabal configure` call.